release notes/v0.37.0.md

```diff
@@ -0,0 +1,61 @@
+
+export default () => {
+    setTimeout(()=> {console.log("say hi to event loops!")}, 200);
+}
+```
+
+## Enhancements and UX improvements
+
+- We continued work on the source map feature: add samples ([#2339](https://github.com/grafana/k6/pull/2339)) and improve loading and parsing ([#2355](https://github.com/grafana/k6/pull/2355)).
+- Updated installation from source instructions ([#2359](https://github.com/grafana/k6/pull/2359)). Thanks, @[karitham](https://github.com/Karitham), for making the change.
+- We migrated our docker repository to the `grafana/k6` ([#2377](https://github.com/grafana/k6/pull/2377)).
+- Introduced a go-based thresholds parsing ([#2400](https://github.com/grafana/k6/pull/2400)).
+- Enabled more TC39 tests: optional-chaining ([#2338](https://github.com/grafana/k6/pull/2338)) and async ([#2396](https://github.com/grafana/k6/pull/2396)).
+
+## Bugs fixed!
+
+- Prevent panic in `check` if a second argument isn't provided ([#2387](https://github.com/grafana/k6/pull/2387)).
+- `http.head` stopped taking a body as a second argument ([#2402](https://github.com/grafana/k6/pull/2402)). A signature looks like how it was before v0.36.0 `http.head(url, [params])`.
+- Fixed `options.scenarios`'s JSON marshaling ([#2392](https://github.com/grafana/k6/pull/2392)).
+- Prevent panic in `http.batch` if pass no arguments (https://github.com/grafana/k6/pull/2415).
+
+## Internals
+
+- We updated our CI to make a developer experience better. Dependency and liter checks run only for pull requests ([#2403](https://github.com/grafana/k6/pull/2403)). We also migrated to a native `golangci-lint` GitHub action. Also bumped the windows version ([#2388](https://github.com/grafana/k6/pull/2388)).
+- Updated goja dependency ([#2350](https://github.com/grafana/k6/pull/2350)).
+- The k6's GRPC module was refactored in order to use the new k6's modules API ([#2365](https://github.com/grafana/k6/pull/2365)) and marked headers parameter as deprecated ([#2370](https://github.com/grafana/k6/pull/2370)).
+- This release also contains a bunch of the refactoring PRs that fix linters errors ([#2334](https://github.com/grafana/k6/pull/2334), [#2331](https://github.com/grafana/k6/pull/2331) and [#2341](https://github.com/grafana/k6/pull/2341)), remove global variables' usage ([#2336](https://github.com/grafana/k6/pull/2336), [#2358](https://github.com/grafana/k6/pull/2358), [#2353](https://github.com/grafana/k6/pull/2353) and [#2357](https://github.com/grafana/k6/pull/2357)) and remove unnecessary dependency ([#2313](https://github.com/grafana/k6/pull/2313)) which makes our codebase more consistent and maintainable.
+- Switched `envconfig` to a fork ([#2337](https://github.com/grafana/k6/pull/2337)) in order to abstract OS package and improve testability.
+- In order to finalize migration to our new API for JS modules, we completely deprecated `common.Bind()` ([#2376](https://github.com/grafana/k6/pull/2376) and [#2384](https://github.com/grafana/k6/pull/2384)).
```

Comment:
I think we should put a bit more emphasis on this, especially if we really plan to drop it in v0.38.0. Every extension that isn't implementing the API will get the warning message, here is an example from `xk6-sql` with master:

```
WARN[0000] Module 'k6/x/sql' is using deprecated APIs that will be removed in k6 v0.38.0, for more details on how to update it see https://k6.io/docs/extensions/guides/create-an-extension/#advanced-javascript-extension
```

I think we should add at least mention what, why and how to fix it, essentially rewriting the same thing we already have in the warning.

Reply:
:ok_hand: Good point, I'll move it out of the bullets section

Reply:
Applied in 3f2b9d2